src/main/drivers/dshot_bitbang_decode.c

```diff
@@ -328,15 +349,26 @@ FAST_CODE uint32_t decode_bb( uint16_t buffer[], uint32_t count, uint32_t bit)
         return DSHOT_TELEMETRY_NOEDGE;
     }
 
-    // The anticipated edges were observed
+    // Data appears valid
+    if (startMargin < minMargin) {
+        minMargin = startMargin;
+    }
 
-    // Attempt to skip the preamble ahead of the telemetry to save CPU
-    if (startMargin > motorConfig()->dev.telemetryStartMargin) {
-        preambleSkip = startMargin - motorConfig()->dev.telemetryStartMargin;
-    } else {
-        preambleSkip = 0;
+    if (cmpTimeUs(now, nextMarginCheckUs)) {
+        nextMarginCheckUs += MARGIN_CHECK_INTERVAL_US;
+
+        if (minMargin > DSHOT_TELEMETRY_START_MARGIN) {
+            preambleSkip = minMargin - DSHOT_TELEMETRY_START_MARGIN;
+        } else {
+            preambleSkip = 20;
+        }
+
+        minMargin = UINT32_MAX;
+
+        DEBUG_SET(DEBUG_DSHOT_TELEMETRY_COUNTS, 3, preambleSkip);
     }
```

Comment:
Repeated code - should we make a function here?

Reply:
Now builds only decode_bb() or decode_bb_bitband() as required.